CMP1903M A01 2223/Pack.cs

```diff
@@ -1,33 +1,143 @@
 {
     class Pack
     {
-        List<Card> pack;
+        public static List<Card> pack = new List<Card>();
+        public static List<Card> tempPack = new List<Card>();
+        public static List<Card> dealtCards = new List<Card>();
 
+        //constructor
         public Pack()
         {
-            //Initialise the card pack here
+
+            for (int suitt = 1; suitt < 5; suitt++)
+            {
+                for (int valuee = 1; valuee < 14; valuee++)
+                {
+                    pack.Add(new Card(valuee, suitt)); //initialising the objects into the list so setting with all the values needed to each object to make a deck of 52 cards
+
+
+                }
+
+
+            }
+
         }
 
+
+
+
         public static bool shuffleCardPack(int typeOfShuffle)
         {
             //Shuffles the pack based on the type of shuffle
 
+            //fisher-yates shuffle
+            if (typeOfShuffle == 1)
+            {
+                Random rnd = new Random();
+                for (int i = pack.Count - 1; i >= 0; i--)
+                {
+                    int Chosen = rnd.Next(0, i);
+                    (pack[Chosen], pack[i]) = (pack[i], pack[Chosen]);
+                }
+                return true;
+            }
+
+            //riffle shuffle
+            else if (typeOfShuffle == 2)
+            {
+
+                int j = 0;//counter
+                while (j < (pack.Count / 2))
+                {
+
+                    tempPack.Add(pack[j]);
+                    tempPack.Add(pack[j + (pack.Count / 2)]);
+                    j += 1;
+
+                }
+                pack = tempPack; //after while loop finishes the temp pack is built with each half of the pack and riffle shuffled and then gets set back to pack 
+
+                return true;
+            }
+
+            //don't shuffle
+            else if (typeOfShuffle == 3)
+            {
+                return true;
+
+
+            }
+            else { return false; }  // exception
+
+
         }
-        public static Card deal()
+
+        //Deals one card
+        public static Card dealCard()
         {
-            //Deals one card
+
+
+            if (pack.Count == 0)
+            {
+                Console.WriteLine("The deck is empty");
+                return null;                     //cant deal a card since there are no cards left so returns null
+
+            }
+            else
+            {
+                Card temp = pack[0];
+                pack.RemoveAt(0);
+                return temp;
+            }
+
 
         }
+        //Deals the number of cards specified by 'amount'
         public static List<Card> dealCard(int amount)
         {
-            //Deals the number of cards specified by 'amount'
+            if (pack.Count == 0)
+            {
+                Console.WriteLine("The deck is empty");
+                return null;
+            }
+
+            else if (pack.Count < amount)
+            {
```

Comment:
Good use of error handling to make sure that there are enough cards in the deck left to be dealt